### DISCUSSION TITLE
[IDEA]: Create separate package for nitro-h3 adapter

### DISCUSSION THREAD

**Discussion by fmoessle** [wobsoriano/trpc-nuxt#94] [Ideas]
Searching for an existing integration with nitro and TRPC lead me here :) because TRPC doesn't support nitro or Nuxt officially.

With great pleasure, I discovered that you implemented a TRPC adapter that not only works for Nuxt but rather is more of a h3-TRPC adapter. The name "createNuxtApiHandler" could even be renamed to "createH3ApiHandler" which would be more appropriate because it is not aware of Nuxt at all.

I want to suggest renaming "createNuxtApiHandler" to "createH3ApiHandler" and/or publishing it separately from this npm package. With this solution people that only want to use nitro and not necessarily Nuxt could benefit from your work!

Thank you for creating this package 🙏

**Comment by wobsoriano** [comment]
Available in 0.10.3 :)

You can import `createH3ApiHandler` from `trpc-nuxt`.

**Reply by michealroberts** [reply]
I've also written this little eventHandler for h3 / nitro -> https://github.com/michealroberts/trpc-nitro-adapter
